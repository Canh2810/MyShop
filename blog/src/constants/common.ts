export const THEMES = {
  DARK: 'dark',
  LIGHT: 'light',
}

export const LOCAL_STORAGE_KEY = {
  AUTH: 'auth',
}

export const LIMIT_POSTS = '9'

export const QUERY_KEY = {
  POSTS: 'posts',
}
