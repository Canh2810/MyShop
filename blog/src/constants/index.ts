export * from './common'
export * from './menu'
export * from './mock-data'
export * from './routes'
