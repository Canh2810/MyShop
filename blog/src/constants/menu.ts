export const LIST_MENU_ITEM_MOCK = [
  { title: 'Home', href: '/' },
  { title: 'About', href: '/about' },
  { title: 'Contact', href: '/contact' },
]

export const HEADER_MENU = [
  { title: 'Home', href: '/' },
  { title: 'Blog', href: '/' },
  { title: 'Single Post', href: '/' },
  { title: 'Pages', href: '/' },
  { title: 'Contact', href: '/' },
]
