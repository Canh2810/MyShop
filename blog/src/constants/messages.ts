export const ERROR_MESSAGES = {
  FIELD_REQUIRED: (fieldName: string) => `${fieldName} is required`,
  FIELD_INVALID: (fieldName: string) =>
    `Invalid format of ${fieldName}, please check.`,
  LOGIN_INVALID: 'Invalid email or password',
  FAIL_TO_FETCH: 'Something went wrong!! Please try again.',
}
