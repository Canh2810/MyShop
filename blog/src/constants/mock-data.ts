export const POST_MOCK = {
  id: 'cvbvbdff3er3',
  category: 'Technology',
  title:
    'The Impact of Technology on the Workplace: How Technology is Changing',
  author: {
    avatar: 'https://www.w3schools.com/w3images/avatar2.png',
    username: 'Tracey Wilson',
  },
  date: 'August 20, 2022',
  imageURL:
    'https://images.unsplash.com/photo-1543877087-ebf71fde2be1?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8OXx8bmF0dXJhbHxlbnwwfHwwfHx8MA%3D%3D&auto=format&fit=crop&w=600&q=60',
}
