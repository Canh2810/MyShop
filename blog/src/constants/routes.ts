export const ROUTES = {
  LOGIN: '/login',
  HOME: '/',
  SINGLE_POST: '/post/:id',
  SINGLE_POST_PARAM: '/post/',
}
