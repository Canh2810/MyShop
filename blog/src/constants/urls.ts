export const BASE_URL = process.env.NEXT_PUBLIC_BASE_URL
export const MOCK_URL = process.env.NEXT_PUBLIC_MOCK_URL
