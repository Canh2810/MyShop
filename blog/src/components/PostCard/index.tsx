'use client'

// Libs
import { memo, useCallback } from 'react'
import Image from 'next/image'
import { useRouter } from 'next/router'

// Types
import { IPost, TypoVariants, CommonVariants } from '@/types'

// Components
import { Badge, Typography } from '../index'

// Constants
import { ROUTES } from '@/constants'

export interface PostCardProps {
  post: IPost
}

const PostCard = ({ post }: PostCardProps) => {
  const { id, title, imageURL, author, category, date } = post || {}
  const { username, avatar } = author || {}
  const router = useRouter()

  // Navigate to single post page
  const handleClickPostCard = useCallback(() => {
    router.push(ROUTES.SINGLE_POST_PARAM + id)
  }, [id, router])

  return (
    <div
      className="p-4 rounded-xl border border-light-200 dark:border-dark-200 border-solid w-fit cursor-pointer"
      onClick={handleClickPostCard}
    >
      <Image
        src={imageURL}
        width={354}
        height={234}
        alt={title}
        style={{ borderRadius: '6px', marginBottom: '16px' }}
        priority
      />
      <div className="p-2 w-[334px] flex flex-col items-start gap-5">
        <Badge variant={CommonVariants.Secondary} title={category} />
        <Typography variant={TypoVariants.HeadingLarge}>{title}</Typography>
        <div className="mt-1 flex items-center gap-5">
          <Image
            src={avatar}
            width={36}
            height={36}
            alt={avatar}
            style={{ borderRadius: '50%', marginRight: '-8px' }}
            priority
          />
          <Typography variant={TypoVariants.TextMedium}>{username}</Typography>
          <Typography variant={TypoVariants.TextMedium}>{date}</Typography>
        </div>
      </div>
    </div>
  )
}

export default memo(PostCard)
