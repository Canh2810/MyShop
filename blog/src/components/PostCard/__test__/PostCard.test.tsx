import { render } from '@testing-library/react'
import PostCard from '..'
import { POST_MOCK, ROUTES } from '@/constants'
import userEvent from '@testing-library/user-event'

const useRouterMock = jest.fn()

jest.mock('next/navigation', () => ({
  ...jest.requireActual('next/navigation'),
  useRouter: () => useRouterMock,
}))

describe('PostCard component', () => {
  it('should render correctly', () => {
    const { container } = render(<PostCard post={POST_MOCK} />)
    expect(container).toMatchSnapshot()
  })

  it('should navigate to the single post page when clicked', () => {
    const routerPushMock = jest.fn()
    useRouterMock.mockReturnValue({ push: routerPushMock })

    const { container } = render(<PostCard post={POST_MOCK} />)
    const postCard = container.querySelector('.post-card')

    if (postCard) {
      userEvent.click(postCard)
      expect(routerPushMock).toHaveBeenCalledWith(
        ROUTES.SINGLE_POST_PARAM + POST_MOCK.id,
      )
    }
  })
})
