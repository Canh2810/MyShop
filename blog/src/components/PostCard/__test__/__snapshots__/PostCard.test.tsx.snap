// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCard component should render correctly 1`] = `
<div>
  <div
    class="p-4 rounded-xl border border-light-200 dark:border-dark-200 border-solid w-fit cursor-pointer"
  >
    <img
      alt="The Impact of Technology on the Workplace: How Technology is Changing"
      data-nimg="1"
      decoding="async"
      fetchpriority="high"
      height="234"
      src="/_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1543877087-ebf71fde2be1%3Fixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxzZWFyY2h8OXx8bmF0dXJhbHxlbnwwfHwwfHx8MA%253D%253D%26auto%3Dformat%26fit%3Dcrop%26w%3D600%26q%3D60&w=750&q=75"
      srcset="/_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1543877087-ebf71fde2be1%3Fixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxzZWFyY2h8OXx8bmF0dXJhbHxlbnwwfHwwfHx8MA%253D%253D%26auto%3Dformat%26fit%3Dcrop%26w%3D600%26q%3D60&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1543877087-ebf71fde2be1%3Fixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxzZWFyY2h8OXx8bmF0dXJhbHxlbnwwfHwwfHx8MA%253D%253D%26auto%3Dformat%26fit%3Dcrop%26w%3D600%26q%3D60&w=750&q=75 2x"
      style="color: transparent; border-radius: 6px; margin-bottom: 16px;"
      width="354"
    />
    <div
      class="p-2 w-[334px] flex flex-col items-start gap-5"
    >
      <span
        class="bg-dark-800 text-primary-500 rounded-md px-2.5 py-1 text-xs font-normal leading-3"
      >
        Technology
      </span>
      <p
        class="font-primary text-xl font-bold leading-6 text-dark-100 dark:text-light-400 "
      >
        The Impact of Technology on the Workplace: How Technology is Changing
      </p>
      <div
        class="mt-1 flex items-center gap-5"
      >
        <img
          alt="https://www.w3schools.com/w3images/avatar2.png"
          data-nimg="1"
          decoding="async"
          fetchpriority="high"
          height="36"
          src="/_next/image?url=https%3A%2F%2Fwww.w3schools.com%2Fw3images%2Favatar2.png&w=96&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fwww.w3schools.com%2Fw3images%2Favatar2.png&w=48&q=75 1x, /_next/image?url=https%3A%2F%2Fwww.w3schools.com%2Fw3images%2Favatar2.png&w=96&q=75 2x"
          style="color: transparent; border-radius: 50%; margin-right: -8px;"
          width="36"
        />
        <p
          class="font-primary text-base font-normal leading-5 text-dark-100 dark:text-light-400 "
        >
          Tracey Wilson
        </p>
        <p
          class="font-primary text-base font-normal leading-5 text-dark-100 dark:text-light-400 "
        >
          August 20, 2022
        </p>
      </div>
    </div>
  </div>
</div>
`;
